stan/math/rev/scal/fun/ibeta.hpp

```diff
@@ -25,7 +26,7 @@ namespace stan {
         double bprod = 1;
         while (std::abs(diff) > precision
                && ++k < max_steps
-               && !std::isnan(diff)) {
+               && !is_nan(diff)) {
```

Comment:
This test is redundant, because if diff is nan, then abs(diff) is nan and the first test will fail.  But no big deal leaving it in if you think it's clearer that way.   And I'd rather see a for loop than this while loop:

```
for (double k = 1; k < max_steps && std::abs(diff) > precision; ++k) {
...
```
but that's a judgement call.

I'd also add space around all those operators and replacd `1.0` with `1`, though the latter's another judgement call about clarity.